File: algorithms/ar-bsrh/java/BinarySearch.java
public class BinarySearch {

    public static void sort(int arr[], int n) {
        int i;
        int j;
        for (i = 0; i < n; i++) {
            for (j = 0; j < n - 1; j++) {
                if (arr[j] >= arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static int binarySearch(int arr[], int low, int high, int searchVal) {
        if (low > high) {
            System.out.println("Search is finished. Value not found");
            return -1;
        }
        int mid = (low + high) / 2;
        if (searchVal == arr[mid]) {
            System.out.println("Search is finished. Index is " + mid);
            return mid;
        } else if (searchVal >= arr[mid]) {
            low = mid + 1;
            return binarySearch(arr, low, high, searchVal);
        } else {
            high = mid - 1;
            return binarySearch(arr, low, high, searchVal);
        }
    }

    public static void main(String[] args) {
        int arr[] = {4, 5, 1, 10, 2, 7, 6, 3};
        sort(arr, arr.length);
        int low = 0;
        binarySearch(arr, low, arr.length - 1, 7);
        binarySearch(arr, low, arr.length - 1, 8);
    }

}

File: algorithms/ar-bsrh/csharp/BinarySearch.cs
// C# implementation of Binary Search 
using System; 

public static class BinarySearch { 
	// Returns index of x if it is present in arr[], 
	// else return -1 
	static int DoBinarySearch(int[] arr, int x) 
	{ 
		int l = 0, r = arr.Length - 1; 
		while (l <= r) {

			int m = l + (r - l) / 2; //mid point

			//1) Check if x is present at mid 
			if (arr[m] == x) 
				{
					return m; 
				}

			//2) If x greater, ignore left half 
			if (arr[m] < x) 
				{
					l = m + 1; 
				}

			//3) If x is smaller, ignore right half 
			else
				{
					r = m - 1; 
				}
		} 

		// if we reach here, then element was 
		// not present 
		return -1; 
	} 

	//Driver program to test above function 
	public static void Main() 
	{ 
		int[] arr = { 2, 3, 4, 10, 40 };
		int x = 10;
		int result = DoBinarySearch(arr, x); 
		if (result == -1) 
		{
			//Element not present"
		}
		else
		{
			//"Element found at index " + result 
		}
		
	} 
} 
